### ISSUE TITLE
[FEA] Apply the new AvroDataFileReader to PERFILE reading of Avro

### ISSUE THREAD

**Issue by firestarman** [NVIDIA/spark-rapids#5458]
This new `AvroDataFileReader ` is introduced in PR https://github.com/NVIDIA/spark-rapids/pull/5421, and can get better perf for cloud reading.
We may want to apply it to Avro PERFILE reading, but this new reader got a little worse perf when reading local files.
```
Local：
      Files     CPU        PERFILE-new   PERFILE-old
      400       25.023     10.172        9.582 
      1000      27.141     16.441        12.722 
      2000      27.179     26.146        21.882 

Cloud：                                     
      128GB     94.134     88.834        830.753 
      1GB       402.612    341.626       3632.736
```
So shall we do this ?

**Comment by firestarman** [comment]
The key point that the new technology can improve perf a lot for cloud is, I think, it reads the data stream directly, without collecting the block metadata first.  So `seek` is called only once at the begining for each file.
But it is called as many times as the block number in collecting block metadata. Then it takes a lot of time, since the `seek` is probably slow for the cloud case.
In my tests, each file is less than 10 MB, so it should read the whole file.

**Comment by firestarman** [comment]
The reasons why the orignal code is faster are, I think,

1. `seek` is fast enough for local files, so collecting metadata should not take much time.
2. With the given block's metadata, most blocks were merged into one copy range, it will reduce the data copying count a lot.
3. It copies the raw data directly into the batch buffer, without parsing.

While the newer code needs to parse the block meta, and then copy only the block data to batch buffer for each block. 

In my tests, each file is about 5 MB, with about 1000 blocks inside. So it should read the whole file.
Besides, the original code will copy the data only once, but the newer code will run data copying about 1000 times.
